src/Uno.UI/UI/Xaml/Media/Animation/ColorAnimationUsingKeyFrames.cs

```diff
@@ -39,16 +39,27 @@ public bool EnableDependentAnimation
 			"KeyFrames",
 			typeof(ColorKeyFrameCollection),
 			typeof(ColorAnimationUsingKeyFrames),
-			new FrameworkPropertyMetadata(default(ColorKeyFrameCollection)));
+			new FrameworkPropertyMetadata(default(ColorKeyFrameCollection), OnKeyFramesChanged));
 		public ColorKeyFrameCollection KeyFrames
 		{
 			get => (ColorKeyFrameCollection)GetValue(KeyFramesProperty);
 			set => SetValue(KeyFramesProperty, value);
 		}
+		private static void OnKeyFramesChanged(DependencyObject sender, DependencyPropertyChangedEventArgs e)
+		{
+			if (sender is ColorAnimationUsingKeyFrames owner)
+			{
+				(e.OldValue as ColorKeyFrameCollection)?.SetParent(null);
+
+				// The parent must always be set, so that items in the collection can use that to walk up the visual tree.
+				// This is needed when updating the resource bindings that point to local resources (for example, from Page.Resources).
+				(e.NewValue as ColorKeyFrameCollection)?.SetParent(owner);
+			}
+		}
 
 		public ColorAnimationUsingKeyFrames()
 		{
-			KeyFrames = new ColorKeyFrameCollection(this, isAutoPropertyInheritanceEnabled: false);
+			KeyFrames = new ColorKeyFrameCollection();
```

Comment:
Shouldn't we still set the parent here, even if it ends up being changed afterwards?